### PR TITLE
UPSE-386: Update to optionally include 'favorite' flag in portlet sea…

### PR DESCRIPTION
**Pull request by groybal**
…rch results

##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]
-   [ ] tests are included
-   [x] new general keys and comments added to `portal.properties`
-   [x] any changes that impact configuration or the database DDL added to `CHANGES.md`

##### Description of change

The portlets search API ( /uPortal/api/v5-0/portal/search?q={search term}*&type=portlets) has been updated to include "favorite" property in the results.

```
{"portlets":[
{"description":"Collection of administrative tools displayed in the ESCO Content Grid","favorite":"false","fname":"admin-dashboard","name":"Admin Dashboard","score":"5.0","title":"Admin Dashboard","url":"/uPortal/f/admin/p/admin-dashboard.u13l1n9/max/render.uP"},{"description":"Administrative functions for tenants in the portal","favorite":"false","fname":"tenant-portal-administration","name":"Tenant Portal Administration","score":"4.0","title":"Tenant Portal Administration","url":"/uPortal/p/tenant-portal-administration.ctf4/max/render.uP"},{"description":"Useful administrative links for managing portal entities and impersonating users","favorite":"false","fname":"portal-administration","name":"Portal Administration","score":"4.0","title":"Portal Administration","url":"/uPortal/p/portal-administration.ctf5/max/render.uP"},
{"description":"Portlet for Portlet Administration","favorite":"true","fname":"portlet-admin","name":"Portlet Administration","score":"4.0","title":"Portlet Administration","url":"/uPortal/f/u15l1s4/p/portlet
```
A new property has been added to enable/disable this:
`org.apereo.portal.rest.search.PortletsSearchStrategy.displayFavoriteFlag`

There was no existing unit test class for PortletsSearchStrategy.  A brief attempt to create one was made, but it was determined that it would take to much time and require some refactoring.

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

File: uPortal-webapp/src/main/resources/properties/portal.properties [hunk 1]
```diff
@@ -545,6 +545,11 @@ org.apereo.portal.portlet.container.services.PortletCookieServiceImpl.portalCook
 ##
 org.apereo.portal.rest.search.PortletsSearchStrategy.displayScore:true
 
+##
+## Flag to enable or disable the display of the favorite flag in the results
+##
+org.apereo.portal.rest.search.PortletsSearchStrategy.favoriteFlagScore:true
```
Comment (groybal) [inline uPortal-webapp/src/main/resources/properties/portal.properties:551]: Hmm, I set this to true but should it instead be 'false' by default?

Comment (jonathanmtran) [reply uPortal-webapp/src/main/resources/properties/portal.properties:551]: Since this is a new property I would be okay with it defaulting to `true`

**Review by cbeach47 — CHANGES_REQUESTED** [review]
Changes look good @groybal ! Leaving the new API config as true works as long as the UX doesn't break with the new field present. Can you confirm the UX works with this new field?

For the tests, can you enhance https://github.com/uPortal-Project/uPortal-start/blob/master/tests/api/search-v5_0.spec.ts with the new field? Even if it's just checking the favorite flag is false, it'd be good. If it's simple/quick, it'd be great to create a new test that favorites a portlet and then confirm the API search call reflect that. Just create a PR in uPortal-start with the change, and once uPortal is released, we can merge in the uPortal-start testing PR.

Note: To run the playwright tests, build uPortal locally, then hook that local build into uPortal-start, and spin up uPortal. Once it's running, then run the uPortal-start gradle task `playwrightRun` .

**Comment by groybal** [comment]
> Changes look good @groybal ! Leaving the new API config as true works as long as the UX doesn't break with the new field present. Can you confirm the UX works with this new field?
> 
> For the tests, can you enhance https://github.com/uPortal-Project/uPortal-start/blob/master/tests/api/search-v5_0.spec.ts with the new field? Even if it's just checking the favorite flag is false, it'd be good. If it's simple/quick, it'd be great to create a new test that favorites a portlet and then confirm the API search call reflect that. Just create a PR in uPortal-start with the change, and once uPortal is released, we can merge in the uPortal-start testing PR.
> 
> Note: To run the playwright tests, build uPortal locally, then hook that local build into uPortal-start, and spin up uPortal. Once it's running, then run the uPortal-start gradle task `playwrightRun` .

I have not noticed any issues in the UX with the new field.
It took some work, but I was able update the Playwright test to check that the new 'favorite' flag is included, and that i has value of true for a favorited portlet.  See PR:  https://github.com/uPortal-Project/uPortal-start/pull/648

**Review by cbeach47 — APPROVED** [review]
LGTM, and thanks for adding the uP-start PW tests!

**Pull request merged**